pkg/clientcert/cert_controller.go

```diff
@@ -258,8 +285,43 @@ func (c *clientCertificateController) sync(ctx context.Context, syncCtx factory.
 		secret.Data = newSecretConfig
 		// save the changes into secret
 		if err := saveSecret(c.spokeCoreClient, c.SecretNamespace, secret); err != nil {
+			if updateErr := c.statusUpdater(ctx, metav1.Condition{
+				Type:    "ClusterCertificateRotated",
+				Status:  metav1.ConditionFalse,
+				Reason:  "ClientCertificateUpdateFailed",
+				Message: fmt.Sprintf("Failed to rotated client certificate %v", err),
+			}); updateErr != nil {
+				return updateErr
+			}
 			return err
 		}
+
+		timeBefore, timeAfter, err := getCertValidityPeriod(secret)
+
+		cond := metav1.Condition{
+			Type:    "ClusterCertificateRotated",
+			Status:  metav1.ConditionTrue,
+			Reason:  "ClientCertificateUpdated",
+			Message: fmt.Sprintf("client certificate rotated starting from %v to %v", *timeAfter, *timeBefore),
```

Comment:
Need to swap timeAfter and timeBefore in the argument list.

Reply:
i thought timeAfter means cert is valid after timeAfter?

Reply:
I checked the code one more time and found `getCertValidityPeriod(secret)` returns `notBefore`, `notAfter` and `error`. So the validity period of a cert should start from `notBefore` to `notAfter`. What do you think?